### DISCUSSION TITLE
How to avoid Text content did not match error with Next.js

### DISCUSSION THREAD

**Discussion by jlak0** [pocketbase/pocketbase#2746] [Q&A]
Get this error after page login ,Next.js pre-rend on the server, 
but in the server side have not my login token so get different page result
How to fix it? I need to send user's credential to the server side and login to get same result in both side?
<img src="https://s2.loli.net/2023/06/19/DYMuJKOdWUBIHcy.png" width=280 height=200 alt=screenshoot ></img>
> `**Unhandled Runtime Error**

>Error: Text content does not match server-rendered HTML.

>Warning: Text content did not match. Server: "please login!!" Client: "hello"

>See more info here: https://nextjs.org/docs/messages/react-hydration-error`

```
"use client";
import PocketBase from "pocketbase";

const pb = new PocketBase("http://localhost:8090");

async function login() {
  const authData = await pb
    .collection("users")
    .authWithPassword("admin", "admin");
  console.log(pb.authStore.isValid);
}
// after the above you can also access the auth data from the authStore

function logout() {
  pb.authStore.clear();
}
// "logout" the last authenticated account
export default function Home() {
  return (
    <div className="flex gap-4">
      {pb.authStore.isValid ? <p>hello</p> : <p>please login!!</p>}
      <button onClick={login}>login</button>
      <button onClick={logout}>logout</button>
    </div>
  );
}
```

**Comment by ganigeorgiev** [comment]
As the provided link mention it is caused by difference between the prerendered page on the server and what is executed in the browser.

The reason why `pb.authStore.isValid` could be different in the 2 environments is that `pb.authStore` by default will try to load any previous auth state from `localStorage` (if available).

What is the end goal here? I don't understand your use case, but if you don't want to use `localStorage` you can provide your own `AuthStore` by extending `BaseAuthStore`, something like:

```js
import PocketBase, { BaseAuthStore } from "pocketbase"

// this auth store will keep the auth data only during the lifetime of the pb instance, aka. node and browser will behave the same
class TempAuthStore extends BaseAuthStore {}

const pb = new PocketBase("http://127.0.0.1:8090", new TempAuthStore());
```

**Reply by ericbaek** [reply]
I am having exact same problem here.
I am well aware that it is caused by difference between the prerendered page on the server and what is executed in the browser.
but can I fix this problem? is there a way to make pre-rendered page same as the page executed in the browser?

**Reply by ganigeorgiev** [reply]
> is there a way to make pre-rendered page same as the page executed in the browser?

You can't make them behave the same because when pre-rendering you may not have access to the browser apis, or as in the above, the browser has a `localStorage` value persisted from some previous run and `pb.authStore` tries to use it, while the server doesn't have access to it. If you haven't already, you can check the error link for more details - https://nextjs.org/docs/messages/react-hydration-error#possible-ways-to-fix-it.

I'm not sure how to help. As mentioned previously I don't understand what is the end goal here. Do you need prerendering? What is the expected behavior if you are fetching a content that is specific to a certain user, how the prerendering is supposed to work in this case? Are you also syncing the auth state server side?

**Reply by ericbaek** [reply]
thanks for trying to help. I've made new discussion for this problem with additional information.
https://github.com/pocketbase/pocketbase/discussions/2846
